src/backend/app/Http/Controllers/Api/v1/ClientsController.php

```diff
@@ -3,32 +3,91 @@
 namespace App\Http\Controllers\Api\v1;
 
 use App\Http\Controllers\Controller;
-use Illuminate\Http\Request;
+use App\Http\Requests\DestroyClient;
+use App\Http\Requests\EditClient;
+use App\Http\Requests\IndexClient;
+use App\Http\Requests\StoreClient;
+use App\Http\Requests\UpdateClient;
+use App\Models\Client;
+use Illuminate\Http\Response;
+use Illuminate\Support\Collection;
 
 class ClientsController extends Controller
 {
-    //TODO: implement me pls
-    public function index(){}
+    /**
+     * Display data of all clients.
+     *
+     * @param IndexClient $request
+     * @return Collection
+     */
+    public function index(IndexClient $request) : Collection{
+        return Client::all();
+    }
 
     //TODO: implement me pls
-    public function create(){}
+    public function create() : Response{
+        return Response::create();
+    }
 
     //TODO: implement me pls
-    public function history(){}
+    public function history(IndexClient $request) : Response{
+        return Response::create();
+    }
 
-    //TODO: implement me pls
-    public function store(){}
+    /**
+     * Store a newly created client in storage.
+     *
+     * @param StoreClient $request
+     * @return JsonResponse
+     */
+    public function store(StoreClient $request) : JsonResponse{
+        $sanitized = $request->validated();
+        $client = Client::create($sanitized);
+        return response()->json($client, 201);
+    }
 
     //TODO: implement me pls
-    public function edit(){}
+    public function edit(EditClient $request, Client $client) : Response{
```

Comment:
aj toto mozes vymazat tu aj z routes, to bolo povodne zamyslane, ze na editaciu bude zvlast podstranka, ale nakoniec @beowulf11 povedal, ze to chce robit inak